doc/src/programmers-guide.md

```diff
@@ -387,6 +399,49 @@ Use `base64:encode/1` and `base64:decode/1` to encode to and decode from Base64
 
 You can Use `base64:encode_to_string/1` and `base64:decode_to_string/1` to perform the same encoding, but to return values as Erlang list structures, instead of as binaries.
 
+### StackTraces
+
+You can obtain information about the current state of a process via stacktraces, which provide information about the location of function calls (possibly including file names and line numbers) in your program.
+
+Current in AtomVM, stack traces can be obtained in one of following ways:
+
+* via try-catch blocks
+* via catch blocks, when an error has been raised via the `error` Bif.
+
+For example a stack trace can be bound to a variable in the catch clause in a try-catch block:
+
+    try
+        do_something()
+    catch
+        _Class:_Error:Stacktrace ->
+            io:format("Stacktrace: ~p~n", [Stacktrace])
+    end
+
+Alternatively, a stack trace can be bound to the result of a `catch` expression, but only when the error is raised by the `error` Bif.  For example,
+
+    {'EXIT', {foo, Stacktrace}} = (catch error(foo)),
+    io:format("Stacktrace: ~p~n", [Stacktrace])
+
+Stack traces are printed to the console in a crash report, for example, when a process dies unexpectedly.
+
+Stacktrace data is represented as a list of tuples, each of which represents a stack “frame”.  Each tuple is of the form:
+
+    [{Module :: module(), Function :: atom(), Arity :: non_neg_integer(), AuxData :: aux_data()}]
+
+where `aux_data()` is a (possibly empty) properties list containing the following elements:
+
+    [{file, File :: binary(), line, Line :: pos_integer()}]
+
+> Note.  The representation of file information as a binary departs from the OTP standard (which uses strings to represent file information), in order to minimize use of heap space on memory-constrained platforms.
```

Comment:
So, I definitely want this optimization for ESP32 platforms, because it seems gratuitous to me to create a list when the data that is needed is in flash.  IMO there should be next to no cases where the file information is needed, programmatically (except perhaps for `io_lib:format`).  As we port to smaller and smaller platforms (like some of the down-sized ESP32 chips), this is going to be more and more important.

Reply:
This departs from the Errors and Error Handling chapter of Erlang's reference manual, especially [this section](https://www.erlang.org/doc/reference_manual/errors#the-call-stack-back-trace--stacktrace-).

It seems like a misplaced memory optimization here. I understand the need to be very strict on memory usage, I also understand that Erlang's usage of lists for strings is historical, as opposed to binaries, yet we would significantly save more RAM by not storing parsed data of modules and reparsing it on demand. Generally, I'm rather in favor of optimizing under constraint rather than proactively with impacts on specifications (and here with not matching Erlang/OTP documentation). This is strictly a recent contributor's opinion, though.

Reply:
I think that providing consistency with Erlang/OTP is very important for developer (and debugging) experience, so I would go with lists even if they are not very efficient.
In case someone is more focused on speed/efficiency, stack traces should be completely disabled.

Reply:
Kept file elements in stacktraces as strings, and optimized to only create one list per stacktrace for each unique module in the stack.  Also added a CMake flag to disable stacktraces, if desired.